AirSkyBoat/AirSkyBoat
# Implement You Call That a Knife Quest

**_I affirm:_**


## Please enter a player-facing description



Implemented the quest "You Call That a Knife?" in Kazham. (HeavensSword)
Implemented the correct dialog for the Tonberry Chef NPCs in Temple of Uggalepih. (HeavensSword)

## What does this pull request do? (Please be technical)



- Created the base quest lua
- Added npc lua stubs for Patissier Linberry and Plongeur Monberry
- Implemented quest logic 
- Added relevant items Sandfish and Tonberry Board to items enum
- Added default actions for NPCs in Kazahm related to the quest: Pula_Rhatti, Vah_Keshura, Mhebi_Juhbily, and _6y9 (Shed)
- Added default actions for Tonberry NPCs in Temple of Uggalepih to provide their proper dialog: Chef_Nonberry, Patissier_Linberry, and Plongeur_Monberry
- Added IDs for "Cook" NMs
- Implemented Tonberry Cook NM spawning from Chef Nonberry
- Chef Nonberry will always consume a non-EX traded item even if he doesn't spawn the NMs
- Created a utility method inside of npc_util.lua that can return true if any items in a trade possess item_flags provided (used for checking for EX items in our use-case)

## Steps to test these changes



- !zone {Kazham}
- Speak with Pula, Vah, Mhebi, and the Shed to verify their base dialogs
- Give yourself Windurst fame >= 6 (may require rezoning after)
- Speak with Pula again and she should mention that Mhebi likes Sandfish
- !additem 4291 (Sandfish)
- Trade the Sandfish to Mhebi and she will ask for it.
- Choose "Forthwith" to continue (or Nope if you want to verify cancelling)
- Speak with the Shed to start a CS
- When prompted, accept the quest and it will be started and added to your quest log
- !zone {Temple of Uggalepih}
- !gotoname Chef_Nonberry
- Speak with Chef_Nonberry, Plongeur Monberry, and Patissier Linberry to verify their base dialog
- Trade an EX item to Chef Nonberry and he should not accept it

NOTE: Spawning these NMs are optional and not required for quest progression.
- Trade any other item to Chef Nonberry and he will say "Chop! Chop!! Chop!!!" and then spawn the NMs: Cook Solberry, Cook Nalberry, Cook Minberry, Cook Fulberry, and Tonberry's Elemental
- If you try trading to Chef Nonberry again while they NMs are spawned or if it has been less than 10 mins since you last spawned, he will not respawn them. He will also still take the item traded! (Retail behavior)

- !additem 1145 (Tonberry Board)
- Trade the Tonberry Board to Chef Nonberry and he will state that he cannot cut it and will give you the Nonberry's Knife KeyItem
- !zone {Kazham}

NOTE: Optional Dialog
- Speak with Vah again and she will be surprised you got the knife

- Speak with Mhebi to complete the quest. You will receive 7200 gil and the Title "Ya Done Good!"
- Speak with Mhebi again for some more dialog
- Zone out and back into Kazham
- Speak with Mhebi once more for her final dialog

## Special Deployment Considerations




In order for the Cook NMs to be spawnable, they must have their spawn positions updated from <0, 0,  0>. Otherwise the system will not load them into the zone mob list.

Run this SQL update:
UPDATE mob_spawn_points SET pos_x = -135.225, pos_y = -1.0, pos_z = -92.232 WHERE mobid >= 17428488 AND mobid <= 17428491
